distributed/tests/test_hdfs.py

```diff
@@ -193,6 +193,50 @@ def test_write_bytes(s, a, b):
 
         assert len(hdfs.ls('/tmp/test/data2/')) == 3
 
+@gen_cluster([(ip, 1), (ip, 1)], timeout=60)
+def test_avro(s, a, b):
+    from distributed.hdfs import avro_to_df
+    import fastavro
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
+    with make_hdfs() as hdfs:
+        # make data
+        schema = {'fields': [{'name': 'key', 'type': 'string'},
+                  {'name': 'value', 'type': 'long'}],
+                  'name': 'AutoGen',
+                  'namespace': 'autogenerated',
+                  'type': 'record'}
+        keys = ("key%s" % s for s in range(10000))
+        vals = range(10000)
+        data = [{'key': key, 'value': val} for key, val in zip(keys, vals)]
+        fnames = ['/tmp/test/1.avro', '/tmp/test/2.avro']
+        for fname in fnames:
+            with hdfs.open(fname, 'w') as f:
+                fastavro.writer(f, schema, data)
+            assert hdfs.info(fname)['size'] > 0
+
+        # simple read
+        fn = fnames[0]
+        with hdfs.open(fn, 'r') as f:
+            av = fastavro.reader(f)
+
+        sync = av._header['sync']
+        by = read_bytes(fn, e, hdfs, False, delimiter=sync, not_zero=True)
+        df = avro_to_df(by[0].result(), av)
```

Comment:
This will hang.  You need to use `_result` here and an explicit `yield`.

Generally I recommend starting with a single very simple test and getting that to pass.  Then you can add more complex tests as you move forward.

For example, it would be really good to test `avro_body` in isolation from all of the HDFS and distributed stuff.  We could also move the avro bytes into a separate variable that are used within a few tests.